components/brave_wallet/common/solana_utils.cc

```diff
@@ -28,10 +28,45 @@ void CompactU16Encode(uint16_t u16, std::vector<uint8_t>* compact_u16) {
   }
 }
 
-bool Base58Decode(const std::string& str, std::vector<uint8_t>* ret, int len) {
+absl::optional<std::tuple<uint16_t, size_t>> CompactU16Decode(
```

Comment:
for reference https://github.com/solana-labs/solana-web3.js/blob/7f33484b2e1160a1bbecb9dc330252d6f7b1a4aa/src/util/shortvec-encoding.ts#L1 and https://docs.rs/solana-program/1.10.10/src/solana_program/short_vec.rs.html#255